packages/graphql-codegen/src/generateGraphqlCodegen.ts

```diff
@@ -1,31 +1,44 @@
-import { EntityDbMetadata, EnumMetadata } from "joist-codegen";
+import { Config, EntityDbMetadata, EnumMetadata } from "joist-codegen";
 import { CodegenFile, code } from "ts-poet";
+import { getEntitiesImportPath } from "./utils";
 
-/** Generates a `graphql-codegen-joist.js` with the auto-generated mapped type/enum value settings. */
-export function generateGraphqlCodegen(entities: EntityDbMetadata[], enums: EnumMetadata): CodegenFile {
+/** Generates a `graphql-codegen-joist.js` (or .mjs for ESM) with the auto-generated mapped type/enum value settings. */
+export function generateGraphqlCodegen(config: Config, entities: EntityDbMetadata[], enums: EnumMetadata): CodegenFile {
   const enumNames = Object.values(enums).map(({ name }) => name);
 
   // Combine the entity mapped types and enum detail mapped types
+  const entitiesMapperImportPath = getEntitiesImportPath(config).replace(/\.ts$/, ""); // Keep compatibility with the expected mapper format (no file extension)
```

Comment:
We will need to dynamically add the `.ts` extension in the resolver codegen plugin. That keeps the mapper definition compatible with the upstream plugins.